app/customer.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from app.car import Car
+from app.point import Point
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict[str, int],
+            location: Point,
+            money: float,
+            car: Car
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart or {}
+        self.location = location
+        self.money = money
+        self.car = car
+
+    @classmethod
+    def create_customer(
+            cls,
+            source: dict[str, str | dict[str, int] | list[int] | int]
```

Comment:
The type hint for `source` should be `dict[str, Union[str, dict[str, int], list[int], float]]` instead of `dict[str, str | dict[str, int] | list[int] | int]` to match the expected types for `name`, `product_cart`, `location`, `money`, and `car`.